recipes/cmake/3.x.x/conanfile.py

```diff
@@ -68,73 +70,73 @@ def validate(self):
             raise ConanInvalidConfiguration(
                 "{} requires a compiler that supports at least C++{}".format(self.name, minimal_cpp_standard))
 
+    def layout(self):
+        if self.options.bootstrap:
+            basic_layout(self, src_folder="src")
+        else:
+            cmake_layout(self, src_folder="src")
+
     def source(self):
-        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-        rmdir(self, os.path.join(self._source_subfolder, "Tests", "RunCMake", "find_package"))
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+        rmdir(self, os.path.join(self.source_folder, "Tests", "RunCMake", "find_package"))
 
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
+    def generate(self):
+        if self.options.bootstrap:
+            tc = AutotoolsToolchain(self)
+            tc.generate()
+            bootstrap_cmake_options = ["--"]
+            bootstrap_cmake_options.append(f'-DCMAKE_CXX_STANDARD={"11" if not self.settings.compiler.cppstd else self.settings.compiler.cppstd}')
+            if self.settings.os == "Linux":
+                if self.options.with_openssl:
+                    openssl = self.dependencies["openssl"]
+                    bootstrap_cmake_options.append("-DCMAKE_USE_OPENSSL=ON")
+                    bootstrap_cmake_options.append(f'-DOPENSSL_USE_STATIC_LIBS={"FALSE" if openssl.options.shared else "TRUE"}')
+                else:
+                    bootstrap_cmake_options.append("-DCMAKE_USE_OPENSSL=OFF")
+            files.save_toolchain_args({"bootstrap_cmake_options": ' '.join(arg for arg in bootstrap_cmake_options)}, namespace="bootstrap")
```

Comment:
same here, `files.save_toolchain_args` may break

Reply:
Replaced with save(..., json.dumps()). FWIW, it's not clear why load & save are part of the Conan API and yet [load|save]_toolchain_args() are not.